src/capi.rs

```diff
@@ -0,0 +1,91 @@
+use std::slice;
+use cose::decoder::decode_signature;
+use cose::SignatureAlgorithm;
+
+unsafe fn from_raw(ptr: *const u8, len: usize) -> Vec<u8> {
+    slice::from_raw_parts(ptr, len).to_vec()
+}
+
+type VerifyCallback = extern "C" fn(*const u8, /* payload */
+                                    usize, /* payload len */
+                                    *const *const u8, /* cert_chain */
+                                    usize, /* # certs */
+                                    *const usize, /* cert lengths in cert_chain */
+                                    *const u8, /* signer cert */
+                                    usize, /* signer cert len */
+                                    *const u8, /* signature bytes */
+                                    usize, /* signature len */
+                                    u8 /* signature algorithm */)
+                                    -> bool;
+
+#[no_mangle]
+pub unsafe extern "C" fn verify_signature_with_cpp(
```

Comment:
Does this whole function need to be unsafe?
Also, technically this is a c api, so maybe verify_signature_from_c? or verify_signature_ffi?

Reply:
`_ffi` sounds good